base/cvd/cuttlefish/host/commands/cvd/server_client.h

```diff
@@ -19,54 +19,101 @@
 #include <sys/socket.h>
 #include <sys/types.h>
 
-#include <optional>
-#include <vector>
+#include <string>
+
+#include <google/protobuf/map.h>
 
 #include "cuttlefish/host/commands/cvd/cvd_server.pb.h"
 
 #include "common/libs/fs/shared_fd.h"
 #include "common/libs/utils/result.h"
-#include "common/libs/utils/unix_sockets.h"
 #include "host/commands/cvd/types.h"
 
 namespace cuttlefish {
 
 class RequestWithStdio {
  public:
-  static RequestWithStdio StdIo(cvd::Request);
-  static RequestWithStdio NullIo(cvd::Request);
-  static RequestWithStdio InheritIo(cvd::Request, const RequestWithStdio&);
+  static RequestWithStdio StdIo();
+  static RequestWithStdio NullIo();
+  static RequestWithStdio InheritIo(const RequestWithStdio&);
 
-  const cvd::Request& Message() const;
   std::istream& In() const;
   std::ostream& Out() const;
   std::ostream& Err() const;
 
   bool IsNullIo() const;
 
-  // Convenient accessors to commonly used properties in the underlying message
-  cvd_common::Args Args() const {
-    return cvd_common::ConvertToArgs(Message().command_request().args());
+  // Convenient getters/setters to properties in the underlying message
+  RequestWithStdio& OverrideRequest(RequestWithStdio) &;
+  RequestWithStdio OverrideRequest(RequestWithStdio) &&;
```

Comment:
What's the difference between these two and `operator=`?

Reply:
The difference is it doesn't reassign the `istream` and `ostream`s. It was only being called in an intermediate commit, so I've removed it.